pyoptsparse/pyOpt_utils.py

```diff
@@ -529,3 +529,13 @@ def _csc_to_coo(mat: dict) -> dict:
     coo_data = np.array(data)
 
     return {"coo": [coo_rows, coo_cols, coo_data], "shape": mat["shape"]}
+
+
+def _broadcast_to_array(name: str, value, n_vars: int, allow_none: bool = True):
```

Comment:
I'm not sure if defaulting `allow_none = True` is sensible, I guess the logic is that by default, `broadcast_to_array` should just broadcast and not do type checking. Happy to change this though.

Reply:
Also not sure if this function should go here. Realistically, this module so far has been a linear algebra library for sparse matrices, so we could move that into a new module and keep things that are truly general utils here. Can also do that separately.

Reply:
1. Sounds good, I will default to `False` and update everything accordingly.
2. I'll leave that as a potential TODO for later then. I'm okay with things as is.